index.bs

```diff
@@ -1549,6 +1549,33 @@ See also:
 
 * [[#naming-is-hard]]
 
+<h3 id="overloading">Use overloading wisely</h3>
+
+If the signature or behaviour of a method will be significantly different
+depending on the arguments passed
+you should define a separate method
+rather than overload a single method.
+
+Overloading a method such that one of its arguments can be
+a single thing or an array is useful.
+Similarly, passing a dictionary of options,
+or using optional arguments,
+are good patterns.
```

Comment:
Using a dictionary of options or optional arguments is not overloading I believe. Are you trying to describe the `addEventListener` case? I can try to reword if so.

Reply:
Optional booleans are generally a smell, though.

Reply:
Also numeric arguments, as @plinss just pointed out in this breakout.